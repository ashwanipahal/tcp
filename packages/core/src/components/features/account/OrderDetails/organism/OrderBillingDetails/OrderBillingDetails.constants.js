const cardIconMapping = {
  DISC: 'disc-small',
  'MASTER CARD': 'mc-small',
  AMEX: 'amex-small',
  VISA: 'visa-small',
  'GIFT CARD': 'gift-card-small',
  'PLACE CARD': 'place-card-small',
  VENMO: 'venmo-blue-acceptance-mark',
  PAYPAL: 'paypal-icon',
};

export default cardIconMapping;
