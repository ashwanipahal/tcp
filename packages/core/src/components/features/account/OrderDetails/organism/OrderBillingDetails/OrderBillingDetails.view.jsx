import React from 'react';
import PropTypes from 'prop-types';
import { BodyCopy } from '@tcp/core/src/components/common/atoms';
import withStyles from '@tcp/core/src/components/common/hoc/withStyles';
import styles from './styles/OrderBillingDetails.style';

/**
 * This function component use for return the EarnPoints
 * can be passed in the component.
 * @param waysToEarn - waysToEarn object used for showing extra points details
 */

const OrderBillingDetails = ({ className }) => {
  /**
   * @function return  Used to render the JSX of the component
   * @param    {[Void]} function does not accept anything.
   * @return   {[Object]} JSX of the component
   */

  return (
    <BodyCopy component="div" className={className}>
      <BodyCopy fontSize="fs14" fontWeight="extrabold" fontFamily="secondary">
        Billing
      </BodyCopy>
    </BodyCopy>
  );
};
OrderBillingDetails.propTypes = {
  className: PropTypes.string,
};

OrderBillingDetails.defaultProps = {
  className: '',
};

export default withStyles(OrderBillingDetails, styles);
export { OrderBillingDetails as OrderBillingDetailsVanilla };
