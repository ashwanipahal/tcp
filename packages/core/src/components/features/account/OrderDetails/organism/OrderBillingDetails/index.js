export { default } from './OrderBillingDetails.view';
