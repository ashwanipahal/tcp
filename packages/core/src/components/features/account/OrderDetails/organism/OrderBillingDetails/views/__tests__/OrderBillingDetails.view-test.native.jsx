import React from 'react';
import { shallow } from 'enzyme';
import OrderBillingDetails from '../OrderBillingDetails.view';

describe('Order Billing Details component', () => {
  it('should renders correctly with credit card details', () => {
    const props = {
      ordersLabels: {},
      className: '',
      orderDetailsData: {
        checkout: {
          billing: {
            card: {
              cardType: 'VISA',
              endingNumbers: '************1111',
            },
          },
        },
        appliedGiftCards: [],
      },
    };
    const component = shallow(<OrderBillingDetails {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('should renders correctly with venmo card ', () => {
    const props = {
      ordersLabels: {},
      className: '',
      orderDetailsData: {
        checkout: {
          billing: {
            card: {
              cardType: 'VENMO',
              endingNumbers: 'testusername',
            },
          },
        },
        appliedGiftCards: [],
      },
    };
    const component = shallow(<OrderBillingDetails {...props} />);
    expect(component).toMatchSnapshot();
  });
});
