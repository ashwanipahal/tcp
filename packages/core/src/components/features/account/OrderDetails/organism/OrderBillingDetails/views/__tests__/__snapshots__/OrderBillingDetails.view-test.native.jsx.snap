// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Billing Details component should renders correctly with credit card details 1`] = `
<Fragment>
  <Styled(View)
    spacingStyles="margin-top-MED"
  >
    <Styled(Styled(BodyCopy))
      accessibilityText=""
      dataLocator=""
      fontFamily="primary"
      fontSize="fs14"
      fontWeight="semibold"
      margin={null}
      numberOfLines={null}
      spacingStyles="margin-top-XS margin-bottom-XS"
      text="lbl_orderDetails_billing"
      textDecoration={null}
    />
    <Styled(View)>
      <Styled(Image)
        source={
          Object {
            "testUri": "../../../../core/src/assets/visa-small.png",
          }
        }
      />
      <Styled(Styled(BodyCopy))
        accessibilityText=""
        dataLocator=""
        fontFamily="primary"
        fontSize="fs12"
        fontWeight="semibold"
        margin={null}
        numberOfLines={null}
        spacingStyles="margin-top-XS margin-bottom-XS"
        text="lbl_orders_ending 1111"
        textDecoration={null}
      />
    </Styled(View)>
  </Styled(View)>
  <Address
    className="elem-mb-SM elem-mt-SM"
    customStyle={Object {}}
    dataLocatorPrefix="address"
    fontSize="fs14"
    regularName={true}
    showCountry={false}
    showDefaultText={false}
    showName={true}
    showPhone={false}
  />
</Fragment>
`;

exports[`Order Billing Details component should renders correctly with venmo card  1`] = `
<Fragment>
  <Styled(View)
    spacingStyles="margin-top-MED"
  >
    <Styled(Styled(BodyCopy))
      accessibilityText=""
      dataLocator=""
      fontFamily="primary"
      fontSize="fs14"
      fontWeight="semibold"
      margin={null}
      numberOfLines={null}
      spacingStyles="margin-top-XS margin-bottom-XS"
      text="lbl_orderDetails_billing"
      textDecoration={null}
    />
    <Styled(View)>
      <Styled(Image)
        source={
          Object {
            "testUri": "../../../../core/src/assets/visa-small.png",
          }
        }
      />
      <Styled(Styled(BodyCopy))
        accessibilityText=""
        dataLocator=""
        fontFamily="primary"
        fontSize="fs12"
        fontWeight="semibold"
        margin={null}
        numberOfLines={null}
        spacingStyles="margin-top-XS margin-bottom-XS"
        text="lbl_orders_ending name"
        textDecoration={null}
      />
    </Styled(View)>
  </Styled(View)>
  <Address
    className="elem-mb-SM elem-mt-SM"
    customStyle={Object {}}
    dataLocatorPrefix="address"
    fontSize="fs14"
    regularName={true}
    showCountry={false}
    showDefaultText={false}
    showName={true}
    showPhone={false}
  />
</Fragment>
`;
